src/modules/trade/hooks/useIsNative.ts

```diff
@@ -0,0 +1,25 @@
+import { Nullish } from 'types'
+
+import { useTokenBySymbolOrAddress } from 'common/hooks/useTokenBySymbolOrAddress'
+
+import { useTradeState } from './useTradeState'
+
+export function useIsNativeIn(): boolean {
```

Comment:
```
function getIsNativeToken(chainId: SupportedChainId, tokenId: string): boolean {
  const nativeToken = NATIVE_CURRENCY_BUY_TOKEN[chainId]

  return checkBySymbolAndAddress(nativeToken, tokenId)
}

export function useIsNativeIn(): boolean {
  const { chainId } = useWalletInfo()
  const { state } = useTradeState()
  const { inputCurrencyId } = state || {}

  return useMemo(() => {
    if (!chainId || !inputCurrencyId) return false

    return getIsNativeToken(chainId, inputCurrencyId)
  }, [chainId, inputCurrencyId])
}
```

Reply:
`useTokenBySymbolOrAddress` hook iterates lots of sources of data, but we have static const `NATIVE_CURRENCY_BUY_TOKEN` and can easily detect a native token using it